### ISSUE TITLE
Question about training stage 3

### ISSUE THREAD

**Issue by nnnnn7896** [sczhou/CodeFormer#171]
Hi, I'm very interesting in your paper and I'm trying to reproduce the training process. I found that in your paper you said stage 3 is to finetune encoder, but in your code the gradiant of encoder features is removed by .detach(). This means encoder will not be updated in stage 3. I'm very confuse about this. Could you help me with this double? By the way, will the decoder also require gradiant during training?

        for i, block in enumerate(self.generator.blocks):
            x = block(x) 
            if i in fuse_list: # fuse after i-th block
                f_size = str(x.shape[-1])
                if w>0:
                    x = self.fuse_convs_dict[f_size](enc_feat_dict[f_size].detach(), x, w)

Thank you for your help!

**Comment by KaKaChan** [comment]
Hello, have you reproduced the training? I think we should remove the ‘detach’ here，But I get bad results so far. I also confuse about this detach ：
        if detach_16:
            quant_feat = quant_feat.detach() # for training stage III

**Comment by KaKaChan** [comment]
Code released！As expected，it involves a lot of manual work to adjust the degradation level and fusion weight for stage 3.
